agent/tcs/handler/handler.go

```diff
@@ -90,6 +83,17 @@ func StartSession(params TelemetrySessionParams, statsEngine stats.Engine) error
 	}
 }
 
+func discoverEndpointAndStartSession(params TelemetrySessionParams, statsEngine stats.Engine) error {
```

Comment:
Can we name this something like `runTelemetrySession`?
The fewer uses of the misnomer 'poll' the better, and I think 'And' is usually awkward in method names.